src/core/fieldformatter/qgsvaluerelationfieldformatter.cpp

```diff
@@ -135,13 +135,19 @@ QgsValueRelationFieldFormatter::ValueRelationCache QgsValueRelationFieldFormatte
     return cache;
 
   QgsFields fields = layer->fields();
-  int ki = fields.indexOf( config.value( QStringLiteral( "Key" ) ).toString() );
-  int vi = fields.indexOf( config.value( QStringLiteral( "Value" ) ).toString() );
+  const int keyIdx = fields.indexOf( config.value( QStringLiteral( "Key" ) ).toString() );
+  const int valueIdx = fields.indexOf( config.value( QStringLiteral( "Value" ) ).toString() );
 
   QgsFeatureRequest request;
 
   request.setFlags( Qgis::FeatureRequestFlag::NoGeometry );
-  QgsAttributeIds subsetOfAttributes { ki, vi };
+  QgsAttributeIds subsetOfAttributes { keyIdx, valueIdx };
+
+  const int groupIdx = fields.indexOf( config.value( QStringLiteral( "Group" ) ).toString() );
```

Comment:
I think these should be lookupField, to be more tolerant when sharing a form config across different data formats:

```suggestion
  const int groupIdx = fields.lookupField( config.value( QStringLiteral( "Group" ) ).toString() );
```